muzzle/src/main/java/io/opentelemetry/javaagent/tooling/ByteArrayUrl.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLEncoder;
+import java.net.URLStreamHandler;
+import java.nio.charset.StandardCharsets;
+import java.security.Permission;
+import java.security.PrivilegedAction;
+
+public class ByteArrayUrl {
+
+  private ByteArrayUrl() {}
+
+  private static final String URL_SCHEMA = "x-otel-binary";
+
+  @SuppressWarnings("removal")
+  public static URL create(String uniqueName, byte[] data) {
+    if (System.getSecurityManager() != null) {
+      return java.security.AccessController.doPrivileged(
+          (PrivilegedAction<URL>)
+              () -> {
+                return doCreate(uniqueName, data);
+              });
+    } else {
+      return doCreate(uniqueName, data);
+    }
+  }
+
+  private static URL doCreate(String uniqueName, byte[] data) {
+    try {
+      String hostname = URLEncoder.encode(uniqueName, StandardCharsets.UTF_8.toString());
+      return new URL(URL_SCHEMA, hostname, -1, "", new ByteArrayUrlStreamHandler(data));
```

Comment:
Using the hostname component instead of file feels a bit weird. Imo using the file would be better, at least then it wouldn't try to do a dns lookup if somehow hashCode or equals is called on this url.

Reply:
This was also simply copied from ByteBuddy. I didn't alter it because it has already been battle-tested there.
If you want to, we can however switch it to the file and see what happens.

Reply:
I still believe that it is better to use the file component rather than the host, it is what everybody else is doing.